### PR TITLE
Implement any and all for pyarrow numpy strings

### PR DESCRIPTION
**Pull request by phofl**
This should work if we want to follow NumPy semantics as close as possible. I think this should work in general, but that's a discussion for another day

### CODE REVIEW

File: pandas/tests/series/test_logical_ops.py [hunk 1]
```diff
@@ -513,3 +513,21 @@ def test_int_dtype_different_index_not_bool(self):
 
         result = ser1 ^ ser2
         tm.assert_series_equal(result, expected)
+
+    def test_any_all(self):
```
Comment (jorisvandenbossche) [inline pandas/tests/series/test_logical_ops.py:517]: It seems there is no other test for any/all in this file. Maybe `pandas/tests/reductions/test_reductions.py` is a better place to put them (or in a file specific for the string arrays)

Comment (phofl) [reply pandas/tests/series/test_logical_ops.py]: moved

**Comment by jbrockmendel** [comment]
> I think this should work in general, but that's a discussion for another day

+1

(Also I recently realized that `Index(["A", "B"]).any_or_all()` is different from `Series(["A", "B"]).any_or_all()` and we should deprecate to get matching behavior)

**Comment by phofl** [comment]
rebased as well

**Comment by lithomas1** [comment]
Is the goal to get this in for 2.1 or is this fine for a later release?

**Comment by phofl** [comment]
Ideally we get this in, but not a deal breaker.

cc @jorisvandenbossche for a final review

**Comment by jorisvandenbossche** [comment]
Opened https://github.com/pandas-dev/pandas/issues/54792 as a general follow-up issue tracking the new string dtype topic.

**Comment by phofl** [comment]
I'll take care of the back-Ort

**Comment by jbrockmendel** [comment]
> I think this should work in general, but that's a discussion for another day

Another day has come!

**Pull request merged**